pageserver/src/virtual_file.rs

```diff
@@ -145,7 +184,9 @@ impl OpenFiles {
         // old file.
         //
         if let Some(old_file) = slot_guard.file.take() {
-            drop(old_file);
+            STORAGE_IO_TIME
+                .with_label_values(&["close", tenantid, timelineid])
+                .observe_closure_duration(|| drop(old_file));
         }
```

Comment:
This accounts the close-time to the *new* tenant and timeline that we are opening, not the file that we are closing.

I wouldn't expect closing files to account for much time, so maybe don't bother tracking the tenant and timeline for closing, and just collect one global metric for that.

Reply:
Sorry, close statistics is really accounts in wrong way.
In principle it is possible to store path together with file or use filepath crate to obtain path for file...
But I agree with you: close is really fast - it is statistic from my notebook:
```
 pageserver_io_time_bucket{operation="close",tenant_id="0a99d78d5aec440ea44517492d8eef45",timeline_id="02d526fcffbe0a985853c8aaf188d2c6",le="0.000001"} 2373
pageserver_io_time_bucket{operation="close",tenant_id="0a99d78d5aec440ea44517492d8eef45",timeline_id="02d526fcffbe0a985853c8aaf188d2c6",le="0.00001"} 4394
pageserver_io_time_bucket{operation="close",tenant_id="0a99d78d5aec440ea44517492d8eef45",timeline_id="02d526fcffbe0a985853c8aaf188d2c6",le="0.0001"} 4487
```
so of close() calls are completed in less than 10 usec. 
I agree that grouping of close metrics by tenant/timelineid may be not needed.
At least for evicted files.I left them for files dropped by user, at least to be able to compare number of manually closed files from evicted files.

Reply:
Ok. A brief code comment to note that we don't track the tenant/timeline for evictions and why would be nice.